### ISSUE TITLE
Tabbing-mode remains active after closing modal

### ISSUE THREAD

**Issue by jesperbrasmussen** [umbraco/Umbraco-CMS#9790]
When a modal is shown (ex. when navigating away from a dirty form) tabbing-mode is activated. 
But if you close the modal without hitting the tab-key, the tabbing-mode remains active. 
Hitting the tab-key while the form is open disables the tabbing-mode when the form is closed.

An side-effect of tabbing-mode beeing enabled is that it result in some strange behaviour for some of our custom datatypes that are using a list of checkboxes. After tabbing-mode has been enabled, you can only change state of the first clicked checkbox. Hitting the tab-key disables the tabbing-mode and you can now check/uncheck the checkboxes at expected.

## Umbraco version

I am seeing this issue on Umbraco version: 8.11.1

I know it wasn't pressent on version 8.6.6

Reproduction
------------
### Specifics

I've reproduced this behavior on Google Chrome using a fresh install of Umbraco 8.11.1

### Steps to reproduce

1. Open a modal by navigating away from a dirty form.
2. click "Discard changes"
3. Tabbing-mode is now still active

![without_tab](https://user-images.githubusercontent.com/1080053/107425751-a87f3800-6b1f-11eb-8fae-3df5e65351c8.gif)

1. Open a modal by navigating away from a dirty form.
2. Press the tab-key at least once
2. click "Discard changes"
3. Tabbing-mode is now disabled

![with_tab](https://user-images.githubusercontent.com/1080053/107425939-ec723d00-6b1f-11eb-8be5-10e37f850246.gif)

### Expected result

I would expect that the tabbing-mode was disabled when I leave a modal even without using the tab-key.

### Actual result

tabbing-mode still active after closing the modal, if tab-key is not pressed

---
_This item has been added to our backlog [AB#11104](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/11104)_

**Comment by emma-hq** [comment]
Hey @jesperbrasmussen 

Thanks for reporting this. Looks like it may indeed be a regression issue. I'll have a look at both 8.6.6 and 8.11.1 to see and if so we'll look at cherry-picking 

Emma

**Comment by emma-hq** [comment]
Hey @jesperbrasmussen 

I'm going to label this one as up-for-grabs - @BatJan, we were wondering if you had any clues as to what had happened here as you've worked on this tabbing,

Emma

**Comment by umbrabot** [comment]
Hi @jesperbrasmussen,

We're writing to let you know that we would love some help with this issue. We feel that this issue is ideal to flag for a community member to work on it. Once flagged here, folk looking for issues to work on will know to look at yours. Of course, please feel free work on this yourself ;-).  If there are any changes to this status, we'll be sure to let you know. 

For more information about issues and states, have a look at [this blog post](https://umbraco.com/blog/the-umbraco-issue-tracker-process/)

Thanks muchly, from your friendly Umbraco GitHub bot :-)

**Comment by BatJan** [comment]
@emma-hq I don't think this is related to the introduction of the focus lock since it sounds like this tabbing mode is set when the modal is closed in a certain way. I don't recall having made any changes to those methods at least 😅 But if time allows I will try to look into it - can probably help fix whether it's something I introduced or not anyway 😁

Ideally the focus should be set to the element that triggered the overlay and then it should be possible to tab back and forth from there on like it was before the modal opened.

**Comment by TwoMoreThings** [comment]
we're seeing an issue that seems to be related to this. For us we aren't touching the tab key, but after a modal is cancelled the first checkbox clicked is the only thing that can be actioned. Clicking anywhere in the dashboard or editor will only action the first checkbox clicked after the dialog is cancelled. We first noticed this in 8.8.0, and it still present in 8.11.2.
HTH
t
If there's a workaround that we could apply it would be very useful at this time.

**Comment by emma-hq** [comment]
Hey @TwoMoreThings 

Just an update for you, Mads is working on a solution to this one in #10074. 

Emma

**Comment by nielslyngsoe** [comment]
Fixed by https://github.com/umbraco/Umbraco-CMS/pull/10074

**Comment by nul800sebastiaan** [comment]
Also cherry picked for patch releases on 8.11 and 8.12.
